custom/icds_reports/utils/aggregation_helpers/agg_awc.py

```diff
@@ -480,29 +511,28 @@ def _transform_column(column_tuple):
                     return column_tuple
                 elif callable(agg_col):
                     return (column, agg_col(column))
-
             return column, 'SUM({})'.format(column)
 
         columns = list(map(_transform_column, columns))
 
-        group_by = ["state_id"]
+        group_by = ["state_id", "state_is_test"]
         if aggregation_level > 1:
-            group_by.append("district_id")
+            group_by.extend(["district_id", "district_is_test"])
         if aggregation_level > 2:
-            group_by.append("block_id")
+            group_by.extend(["block_id", "block_is_test"])
         if aggregation_level > 3:
-            group_by.append("supervisor_id")
+            group_by.extend(["supervisor_id", "supervisor_is_test"])
```

Comment:
Grouping by `is_test` seems a bit weird to me. I'd have thought you'd want `max(supervisor_is_test)` which will tell you if  the supervisor was ever 'test' in the time period.

Reply:
@snopoke True, but we aggregate data for a specific month and calculate data by location below, so for example when we collecting data for supervisor we take data for the specific month for all children. I think we can't have a possibility that one child will have ```is_test``` for parent ```True``` and one ```False```